iceoryx_utils/source/posix_wrapper/semaphore.cpp

```diff
@@ -201,30 +229,53 @@ Semaphore::Semaphore(iox_sem_t* handle, const unsigned int value) noexcept
     {
         m_isInitialized = true;
     }
+    else
+    {
+        m_isInitialized = false;
+        m_errorValue = SemaphoreError::CREATION_FAILED;
+    }
 }
 
-Semaphore::Semaphore(const char* name, const int oflag) noexcept
+Semaphore::Semaphore(OpenNamedSemaphore_t, const char* name, const int oflag) noexcept
```

Comment:
The Fixedstring should be used for the name to avoid unsafe strncpy operation below.

Reply:
The argument is forwarded to `sem_open` which is using a `const char *`, see: https://www.man7.org/linux/man-pages/man3/sem_open.3.html

Therefore, if we run into issues `smart_c` will catch them and inform the user via the creation pattern.

The fixed string should be used in the upper layers.